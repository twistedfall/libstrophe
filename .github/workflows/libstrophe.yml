name: libstrophe
on:
  push:
    branches:
      - "*"
    tags_ignore:
      - "*"
  pull_request:
    branches:
      - "*"
jobs:
  ci:
    strategy:
      fail-fast: false
      matrix:
        libstrophe-version:
          - 0.9.3
          - 0.10.1
          - 0.11.0
          - 0.12.3
          - 0.13.1
          - 0.14.0
    runs-on: ubuntu-24.04
    env:
      LIBSTROPHE_VERSION: ${{ matrix.libstrophe-version }}
      SCCACHE_GHA_ENABLED: "true"
      RUSTC_WRAPPER: "sccache"
    steps:
      - uses: actions/checkout@v4
      - uses: mozilla-actions/sccache-action@v0.0.9
      - uses: dtolnay/rust-toolchain@stable
        with:
          components: clippy,rustfmt

      - run: ci/install.sh
        shell: bash

      - run: ci/script.sh
        shell: bash

  msrv:
    runs-on: ubuntu-24.04
    env:
      SCCACHE_GHA_ENABLED: "true"
      RUSTC_WRAPPER: "sccache"
    steps:
      - uses: actions/checkout@v4
      - uses: mozilla-actions/sccache-action@v0.0.9
      - uses: dtolnay/rust-toolchain@master
        with:
          toolchain: ${{ steps.metadata.outputs.msrv }}
          components: clippy,rustfmt

      - name: Get MSRV package metadata
        id: metadata
        run: cargo metadata --no-deps --format-version 1 | jq -r '"msrv="+(.packages[]|select(.name=="libstrophe").rust_version)' >> $GITHUB_OUTPUT

      - run: ci/checks.sh
        shell: bash
