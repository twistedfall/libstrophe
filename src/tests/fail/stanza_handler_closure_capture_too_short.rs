use libstrophe::{Context, Connection};

fn main() {
	{
		let mut val = 0;
		let mut conn = Connection::new(Context::new_with_null_logger());
		conn.handler_add(|_, _, _| {
			val = 1;
			false
		}, None, None, None);
		conn
	};
}
