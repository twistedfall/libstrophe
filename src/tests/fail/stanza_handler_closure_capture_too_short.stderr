error[E0597]: `val` does not live long enough
  --> src/tests/fail/stanza_handler_closure_capture_too_short.rs:9:5
   |
5  |         let mut val = 0;
   |             ------- binding `val` declared here
...
8  |             |_, _, _| {
   |             --------- value captured here
9  |                 val = 1;
   |                 ^^^ borrowed value does not live long enough
...
16 |         conn
   |         ---- a temporary with access to the borrow is created here ...
17 |     };
   |     -- ... and the borrow might be used here, when that temporary is dropped and runs the `Drop` code for type `Connection`
   |     |
   |     `val` dropped here while still borrowed
   |
help: consider adding semicolon after the expression so its temporaries are dropped sooner, before the local variables declared by the block are dropped
   |
16 |         conn;
   |             +
